ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/SubordinateTransaction.java

```diff
@@ -113,4 +116,28 @@ public void doOnePhaseCommit () throws IllegalStateException,
     public Xid baseXid();
 
     public Uid get_uid();
+
+    /**
+     * <p>
+     * Listing deferred throwables gathered during the transaction processing.
+     * This list contains exceptions thrown by transaction resources when an error
+     * occurs.
+     * <p>
+     * Depending on the transaction stage the particular method returns only an error code e.g. {@link #doPrepare()}
+     * then the controller of the subordinate transaction throws its own exception and it could
+     * pack these throwables in some details.
+     * Other methods as {@link #doCommit()} throws an exception but those exception does not contain
+     * underlying exceptions like those coming from resource failures.
+     * <p>
+     * Normally exceptions taken by this call are put under suppressed part of the thrown exception.
+     *
+     * <p>
+     * JTS implementation of the {@link SubordinateTransaction} interface does not support the deferred throwable
+     * based on the limitation of ORB protocol. This method called on JTS implementation causes {@link UnsupportedOperationException}
+     * being thrown.
+     *
+     * @return list of throwables causing the failure state of the transaction
+     * @throws UnsupportedOperationException for JTS implementation
+     */
+    public List<Throwable> getDeferredThrowables();
```

Comment:
I added the `supportsDeferredThrowables` and used in EAP 6.4 ejb subsystem fix.

The ejb subsystem runs directly with the `SubordinateTransaction` as call `SubordinationManager.getTransactionImporter()..getImportedTransaction(Xid)`. From EAP 6.4 point I think it's not needed this methods being added under `ImportedTransaction` but if you think it should be there I can try to investigate. Should I?

By quick checking the WFTC I think it uses the simlar approach and direct calls to `SubordinateTransaction` too. But it needs to be confirmed by @dmlloyd .